sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -94,54 +96,74 @@ object JDBCRDD extends Logging {
     new StructType(columns.map(name => fieldMap(name)))
   }
 
-  /**
-   * Turns a single Filter into a String representing a SQL expression.
-   * Returns None for an unhandled filter.
-   */
-  def compileFilter(f: Filter, dialect: JdbcDialect): Option[String] = {
-
-    def quote(colName: String): String = {
-      val nameParts = SparkSession.active.sessionState.sqlParser.parseMultipartIdentifier(colName)
-      if(nameParts.length > 1) {
-        throw QueryCompilationErrors.commandNotSupportNestedColumnError("Filter push down", colName)
-      }
-      dialect.quoteIdentifier(nameParts.head)
+  private def checkColumnName(colName: String) = {
+    val nameParts = SparkSession.active.sessionState.sqlParser.parseMultipartIdentifier(colName)
+    if(nameParts.length > 1) {
+      throw QueryCompilationErrors.commandNotSupportNestedColumnError("Filter push down", colName)
     }
+  }
 
+  private[sql] def toV2(f: Filter): Option[Predicate] = {
```

Comment:
shall we put this function in v1 `Filter`?

Reply:
v1 `Filter` in catalyst and `SparkSession` in core.

Reply:
Can't we use `CatalystSqlParser`?

Reply:
I moved `toV2` to `JDBCRelation`.